machiavelli-cards/src/main/java/pl/jellytech/machiavelli/cards/controllers/CardController.java

```diff
@@ -0,0 +1,96 @@
+package pl.jellytech.machiavelli.cards.controllers;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+import pl.jellytech.machiavelli.cards.dtos.CardResponse;
+import pl.jellytech.machiavelli.cards.entities.Card;
+import pl.jellytech.machiavelli.cards.entities.CardType;
+import pl.jellytech.machiavelli.cards.services.CardService;
+import pl.jellytech.machiavelli.cards.utils.ControllerUtils;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@RestController()
+@RequestMapping( "api/card")
+@Slf4j
+public class CardController {
+    private final CardService cardService;
+    @Autowired
+    public CardController(CardService cardService){
+        this.cardService = cardService;
+    }
+
+    @PostMapping(consumes ={MediaType.MULTIPART_FORM_DATA_VALUE} )
+    public ResponseEntity createOrUpdate(@RequestPart("cardType") String cardType,
+                     @RequestPart("name") String name,
```

Comment:
remind me again why we don't just accept a Card object?

Reply:
We have to upload card image. So i have used multipart form-data payload. I haven't find any solution to bind on single dto object

Reply:
I'll see if I can find examples - ok for now - will log this as issue to improve later on